pkg/cloudinit/common.go

```diff
@@ -73,7 +73,7 @@ func NewBaseCloudConfig(data BaseUserData) (CloudConfig, error) {
 
 	snapInstall := data.SnapInstallData
 	// Default to k8s version if snap install option is not set or empty.
-	if snapInstall == nil {
+	if snapInstall == nil || snapInstall.Option == "" {
 		snapInstall = &SnapInstallData{
 			Option: InstallOptionChannel,
 			Value:  fmt.Sprintf("%d.%d-classic/stable", kubernetesVersion.Major(), kubernetesVersion.Minor()),
```

Comment:
edge case nit: What happens if the user selects a Kubernetes version where we do not (yet) have a stable release? Do we provide a good error experience for that?